be/src/storage/column_predicate_rewriter.cpp

```diff
@@ -48,46 +48,121 @@ namespace starrocks {
 
 constexpr static const LogicalType kDictCodeType = TYPE_INT;
 
-Status ColumnPredicateRewriter::rewrite_predicate(ObjectPool* pool) {
-    // because schema has reordered
-    // so we only need to check the first `predicate_column_size` fields
-    for (size_t i = 0; i < _column_size; i++) {
-        const FieldPtr& field = _schema.field(i);
-        const auto cid = field->id();
+struct RewritePredicateTreeVisitor {
+    using RewriteStatus = ColumnPredicateRewriter::RewriteStatus;
+
+    template <CompoundNodeType ParentType>
+    StatusOr<RewriteStatus> operator()(PredicateColumnNode& node, PredicateCompoundNode<ParentType>& parent) const {
+        const auto* col_pred = node.col_pred();
+        const auto cid = col_pred->column_id();
 
-        auto iter = _pred_map.find(cid);
-        if (iter == _pred_map.end()) {
-            continue;
+        if (!_rewriter._need_rewrite[cid]) {
+            return RewriteStatus::UNCHANGED;
         }
-        auto& preds = iter->second;
 
-        std::vector<const ColumnPredicate*> remove_list;
-        for (auto& pred : preds) {
-            ColumnPredicate* new_pred = nullptr;
-            ASSIGN_OR_RETURN(auto rewrite_status, _rewrite_predicate(pool, field, pred, &new_pred));
+        const auto& field = _cid_to_field.find(cid)->second;
+        DCHECK(_rewriter._column_iterators[cid]->all_page_dict_encoded());
```

Comment:
use a function instead of field

Reply:
`RewritePredicateTreeVisitor` is just a part implementation of `ColumnPredicateRewriter::rewrite_predicate`. This is similar to IMPL. Therefore, I think it is ok to access private members of `ColumnPredicateRewriter`.